packages/flutter_tools/lib/src/version.dart

```diff
@@ -73,14 +98,18 @@ class FlutterVersion {
 
   String _runGit(String command) => runSync(command.split(' '), workingDirectory: Cache.flutterRoot);
 
+  void ensureVersionFile() {
```

Comment:
Should this just be written out during construction?

Reply:
If it does make sense to keep this as a separate method called by the command runner, then lat's make it async.

Reply:
i'd rather not have a constructor that mutates the world (notwithstanding that it already runs subprocesses and stuff, i just don't want to make it worse...). But I've made it async.